cluster_test.go

```diff
@@ -1603,6 +1606,24 @@ func (s *S) TestPoolLimitSimple(c *C) {
 		session.Refresh()
 		delay := <-done
 		c.Assert(delay > 300*time.Millisecond, Equals, true, Commentf("Delay: %s", delay))
+		stats := mgo.GetStats()
+		c.Assert(stats.TimesSocketAcquired, Equals, 2)
+		c.Assert(stats.TimesWaitedForPool, Equals, 1)
+		c.Assert(stats.PoolTimeouts, Equals, 0)
+		c.Assert(stats.TotalPoolWaitTime > 300*time.Millisecond, Equals, true)
+		statsEvents := mgo.GetStatsEvents()
+		// Should be two acquisition events; one from the first .Ping(), and one
+		// from the goroutine
+		getEvent := func() *mgo.SocketAcquireMetric {
+			select {
+			case ev := <-statsEvents.OnSocketAcquired:
+				return &ev
+			default:
+				return nil
+			}
+		}
+		c.Assert(getEvent().WaitTime, Equals, time.Duration(0))
```

Comment:
This is a nitpick, but I would check if the result of getEvent() is not nil before accessing the WaitTime. 
If all works well that's not a problem, but if in subsequent commits someone breaks the stats by chance, this will panic.  And failing unit test is much easier to deal with than a panicking one (imho).